### PR TITLE
Feature new/null database fields csmapis

### PR DESCRIPTION
**Pull request by NickyDaB**
**Summary:**
- Adds in the ability to reset fields in the CSM Database to NULL via a csm API. 
- Reset a field to NULL with the special keyword "#CSM_NULL"
- designed for strings and text fields
- new function added in to do the comparing `CAST_stringTools_CSM_KEYWORD_Compare(input->comment, &keyword_compareCode)`
- this compares the string to all the CSM keywords we currently have
- it takes in the string, and a reference to an int, which the function uses to give the user some feedback.
- function is located in the `utilities/include/string_tools.h` file
- There is a RAS API that also resets a field back to NULL, but that wasn't a string; we use an ENUM in that case. 

**Reason behind Pull Request:**
Support for this feature was requested by the labs. Original issue can be found here: https://github.com/IBM/CAST/issues/108

### CODE REVIEW

File: utilities/src/string_tools.c [hunk 1]
```diff
@@ -247,3 +247,79 @@ int CORAL_stringTools_nodeCount_xCATSyntax(char* myString, int* dataCount)
...
+*   1 keyword detected but no known match
+*   2 #CSM_NULL
+*
+* Returns: int 
+*   0 Success
+*   1 ERROR: Generic error. Default.
+*   2 ERROR: myString is NULL. Can not parse NULL string.
+*   3 ERROR: reached end of program without returning valid data
+*/
+int CAST_stringTools_CSM_KEYWORD_Compare(char* myString, int* compareCode)
```
Comment (mew2057) [inline utilities/src/string_tools.c]: We should probably remove this with an enumerated type in the future. (Low Priority)

File: csmd/src/daemon/src/csmi_request_handler/CSMISwitchAttributesUpdate.cc [hunk 2]
```diff
@@ -36,64 +36,113 @@ bool CSMISwitchAttributesUpdate::CreatePayload(
...
+	bool comment_NULL = false;
+
+	std::string stmt = "WITH updated AS ( UPDATE csm_switch SET ";
+	
+	//helper for keyword compare.
+	int keyword_returnCode = 0;
+	int keyword_compareCode = 0;
+	
+	//check to see if we have to do anything for comment. 
+	if(input->comment[0] != '\0')
```
Comment (mew2057) [inline csmd/src/daemon/src/csmi_request_handler/CSMISwitchAttributesUpdate.cc]: Is there a way to reduce the LOC for this, maybe make it more function driven? (Low priority)

Comment (NickyDaB) [reply csmd/src/daemon/src/csmi_request_handler/CSMISwitchAttributesUpdate.cc]: I thought about it. Almost made a macro, but I decided to go with this for future "edit-ability". This way if someone needs to come in and add specific functionality, then it is already all set up. And they can just insert the functionality code and not have to write logic code.

We could reduce the switch statement down, or marco-ise it, but then if someone wants to add in that extra functionality they would need to spend the extra time to expand it out. Based off of previous development cycles and requests for features on this project. I think this is the best move to mitigate future hassle. 

Here the switch statement already includes current possible outputs with comments describing each number and what they mean, so a new programmer on the project wouldn't have to explore code to look up the function either. 

But maybe there is a way to reduce lines without impact usability for the future programmer. I didn't think of any at the moment.

File: utilities/src/string_tools.c (hunk 1 above)
Comment (NickyDaB) [reply utilities/src/string_tools.c]: I agree. In the future we should have a file or list of all the `CSM_KEYWORDS` which are enums. I'll create an issue for it.

**Comment by NickyDaB** [comment]
merging in

**Pull request merged**